services/wallet/service.go

```diff
@@ -477,3 +519,70 @@ func (service *Service) DisconnectCustodianLink(ctx context.Context, custodian s
 	}
 	return nil
 }
+
+// CreateBraveWallet creates a brave rewards wallet and informs the reputation service.
+// If either the local transaction or call to the reputation service fails then the wallet is not created.
+func (service *Service) CreateBraveWallet(ctx context.Context, publicKey string, geolocation string) (*walletutils.Info, error) {
+	valid, err := service.geolocationValidator.Validate(ctx, geolocation)
+	if err != nil {
+		return nil, fmt.Errorf("error validating geolocation: %w", err)
+	}
+
+	if !valid {
+		return nil, errGeoLocationDisabled
+	}
+
+	var altCurrency = altcurrency.BAT
+	var info = &walletutils.Info{
+		ID:          uuid.NewV5(ClaimNamespace, publicKey).String(),
+		Provider:    "brave",
+		PublicKey:   publicKey,
+		AltCurrency: &altCurrency,
+	}
+
+	ctx, tx, rollback, commit, err := getTx(ctx, service.Datastore)
+	if err != nil {
+		return nil, fmt.Errorf("error creating transaction: %w", err)
+	}
+	defer rollback()
+
+	err = service.Datastore.InsertWalletTx(ctx, tx, info)
```

Comment:
ctx has the tx on it at this point, if we go ahead and make insert wallet method use getTx.